routers/repo/view.go

```diff
@@ -103,13 +103,54 @@ func renderDirectory(ctx *context.Context, treeLink string) {
 		isTextFile := base.IsTextFile(buf)
 		ctx.Data["FileIsText"] = isTextFile
 		ctx.Data["FileName"] = readmeFile.Name()
+		fileSize := int64(0)
+		isLFSFile := false
+		ctx.Data["IsLFSFile"] = false
+
 		// FIXME: what happens when README file is an image?
 		if isTextFile {
-			if readmeFile.Size() >= setting.UI.MaxDisplayFileSize {
+			if setting.LFS.StartServer {
+				if meta := lfs.IsPointerFile(&buf); meta != nil {
```

Comment:
This could be better restructured to not have so much levels:
```
meta := lfs.IsPointerFile(&buf)
if meta != nil {
  meta, _ = ctx.Repo.Repository.GetLFSMetaObjectByOid(meta.Oid)
}
if meta != nil {
  ctx.Data["IsLFSFile"] = true
  ...
}
```

Reply:
also shouln't we at least log warning if `GetLFSMetaObjectByOid` returns error?

Reply:
done.

Reply:
ctx.ServerError will log on error